octree_web_viewer/client/main.ts

```diff
@@ -20,124 +20,203 @@ import { FirstPersonController } from './control';
 import { OctreeViewer } from './octree_viewer';
 
 class App {
-  private camera: THREE.PerspectiveCamera;
-  private scene: THREE.Scene;
-  private controller: FirstPersonController;
-  private viewer: OctreeViewer;
-  private renderer: THREE.WebGLRenderer;
-  private lastFrustumUpdateTime: number;
-  private lastMoveTime: number;
-  private needsRender: boolean;
-
-  public run() {
-    let renderArea = document.getElementById('renderArea');
-    this.renderer = new THREE.WebGLRenderer();
-    this.renderer.setSize(renderArea.clientWidth, renderArea.clientHeight);
-    renderArea.appendChild(this.renderer.domElement);
-
-    const VIEW_ANGLE = 45;
-    const NEAR = 0.1;
-    const FAR = 10000;
-    this.camera = new THREE.PerspectiveCamera(
-      VIEW_ANGLE,
-      renderArea.clientWidth / renderArea.clientHeight,
-      NEAR,
-      FAR
-    );
-    this.camera.position.z = 150;
-    this.camera.updateMatrix();
-    this.camera.updateMatrixWorld(false);
-
-    this.scene = new THREE.Scene();
-    this.scene.add(this.camera);
-
-    this.controller = new FirstPersonController(
-      this.camera,
-      this.renderer.domElement
-    );
-
-    this.lastFrustumUpdateTime = 0;
-    this.lastMoveTime = 0;
-    this.needsRender = true;
-    this.viewer = new OctreeViewer(this.scene, () => {
-      this.needsRender = true;
-    });
-    const gui = new GUI();
-    gui
-      .add(this.viewer.material.uniforms['size'], 'value')
-      .name('Point size')
-      .onChange(() => {
-        this.needsRender = true;
-      });
-    gui
-      .add(this.viewer.material.uniforms['alpha'], 'value', 0, 1)
-      .name('Transparency')
-      .onChange(() => {
-        this.viewer.alphaChanged();
-        this.needsRender = true;
-      });
-    gui
-      .add(this.viewer.material.uniforms['gamma'], 'value')
-      .name('Gamma')
-      .onChange(() => {
-        this.needsRender = true;
-      });
-    gui
-      .add(this.viewer, 'maxLevelToDisplay', 0, 7)
-      .name('Moving details')
-      .step(1)
-      .onChange(() => {
+    private camera: THREE.PerspectiveCamera;
+    private scene: THREE.Scene;
+    private controller: FirstPersonController;
+    private viewer: OctreeViewer;
+    private renderer: THREE.WebGLRenderer;
+    private lastFrustumUpdateTime: number;
+    private lastMoveTime: number;
+    private needsRender: boolean;
+    private octreeIdControl: dat.GUIController;
+    private gui: dat.GUI;
+    private guiRenderControls: dat.GUI;
+    public octreeId: string;  // octree identifier
+
+    private fetchDefaultOctreeId(): Promise<string> {
+        const request = new Request(
+            `/init_tree`,
+            {
+                method: 'GET',
+                credentials: 'same-origin',
+            }
+        );
+
+        const result = window
+            .fetch(request)
+            .then((response) => response.text()) // todo error handling?
```

Comment:
Can you please reformat to our default TODO style: `TODO(cvitadello): Error handling`?

Reply:
Does this imply the she would be the future assignee?

Reply:
If you already know, who is supposed to do the work, we should put her name in here...

Reply:
I have been working on a forked version of this component. It can be assigned to me, though I don't have the capacity for getting to it at the moment.

Reply:
If you agree negin, i'd put your name on that because you have a better understanding of what it has to be done in the future.